orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -85,7 +85,7 @@ class RunTaskHandler(
                 }
                 TERMINAL, CANCELED                   -> {
                   val status = stage.failureStatus(default = result.status)
-                  queue.push(CompleteTask(message, status))
+                  queue.push(CompleteTask(message, status, result.status))
```

Comment:
Yeah, it feels like the `failureStatus` calculation would make more sense in `CompleteTaskHandler`.

Reply:
Agreed ... I think we still need to do a pass (like this) where we're supplying the explicit value separately to remain backwards compatible. 

Then we can change `CompleteTaskHandler` to only look at the original status ... then we can set status == originalStatus ... then we can remove originalStatus.

A pain but I think this will migrate us without significant issue.

Reply:
Or maybe we can just take a shortcut and do it all in one swoop.

Reply:
Oh, I see. Didn't realize that's what you intended. Given the infrequency of stages returning that is it really worth the effort?